### PR TITLE
Load of small fixes to reduce unused code in utils

### PR DESCRIPTION
**Pull request by EmmanuelMess**
* Deleted useless lines (empty javadoc or `@return` or `@param` without explanation in docs)
* Deleted unused methods
* Simplified code where possible
* Simplified MapEntry with Guava's ImmutableEntry 
* Deleted Logger
* Renamed RootNotPermittedException to ShellNotRunningException
* Deleted CryptException and replace with GeneralSecurityException and IOException

### CODE REVIEW

File: app/src/main/java/com/amaze/filemanager/utils/RootUtils.java [hunk 1]
```diff
@@ -109,195 +68,124 @@ private static String mountFileSystemRW(String path) throws RootNotPermittedExce
...
 
         RootHelper.runShellCommand("cat \"" + sourcePath + "\" > \"" + destinationPath + "\"");
-        if (mountPoint!=null) {
+        if (mountPoint != null) {
             // we mounted the filesystem as rw, let's mount it back to ro
             mountFileSystemRO(mountPoint);
         }
     }
 
-    public static String parsePermission(String permLine) {
```
[the remark this answers is not in the document]
Comment (EmmanuelMess) [reply app/src/main/java/com/amaze/filemanager/utils/RootUtils.java]: How?

[the remark this answers is not in the document]
Comment (VishalNehra) [reply app/src/main/java/com/amaze/filemanager/utils/RootUtils.java]: What do you mean by, how? Just keep this method and replace the code in ````GeneralDialogCreation#showPermissionsDialog```` which is related to changing permissions with this method callback.

[the remark this answers is not in the document]
Comment (EmmanuelMess) [reply app/src/main/java/com/amaze/filemanager/utils/RootUtils.java]: [GeneralDialogCreation#showPermissionsDialog](https://github.com/TeamAmaze/AmazeFileManager/blob/438d6213702aa3e89cf69a8a96b3001a7f92fe34/app/src/main/java/com/amaze/filemanager/ui/dialogs/GeneralDialogCreation.java#L910) needs to convert from checkboxes to octal (actually String, not int), this doesn't help...

File: app/src/main/java/com/amaze/filemanager/utils/RootUtils.java [hunk 2]
```diff
@@ -21,55 +21,15 @@
...
-        return arrayLis;
-    }
-
-    /**
-     * Change permissions (owner/group/others) of a specified path
-     * @param path
-     * @param octalNotation octal notation of permission
-     * @throws RootNotPermittedException
-     */
-    public static void chmod(String path, int octalNotation) throws RootNotPermittedException {
```
[the remark this answers is not in the document]
Comment (EmmanuelMess) [reply app/src/main/java/com/amaze/filemanager/utils/RootUtils.java]: This method is unusable: it requires octal int, doesn't return result and doesn't allow for options. I will create a variable `CHMOD_COMMAND = "chmod %s \"%s\""` that should be used with `String.format()` e.g. `String.format(CHMOD_COMMAND, "-R", "/a/", "741");`.

File: app/src/main/java/com/amaze/filemanager/utils/RootUtils.java [hunk 3]
```diff
@@ -109,195 +68,124 @@ private static String mountFileSystemRW(String path) throws RootNotPermittedExce
...
 
-
-    /**
-     * Returns file permissions in octal notation
-     * Method requires busybox
-     * @param path
-     * @return
-     */
-    private static int getFilePermissions(String path) throws RootNotPermittedException {
-        String line = RootHelper.runShellCommand("stat -c  %a \"" + path + "\"").get(0);
```
[the remark this answers is not in the document]
Comment (EmmanuelMess) [reply app/src/main/java/com/amaze/filemanager/utils/RootUtils.java]: Done.

**Pull request merged**